pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/MultiStageEngineIntegrationTest.java

```diff
@@ -704,6 +704,26 @@ public void testVariadicFunction() throws Exception {
     assertEquals(jsonNode.get("numRowsResultSet").asInt(), 3);
   }
 
+  @Test
+  public void skipArrayToMvOptimization()
+      throws Exception {
+    String sqlQuery = "SELECT 1 "
+        + "FROM mytable "
+        + "WHERE ARRAY_TO_MV(RandomAirports) = 'MFR' and ARRAY_TO_MV(RandomAirports) = 'GTR'";
+
+    JsonNode jsonNode = postQuery("Explain plan for " + sqlQuery);
+    JsonNode plan = jsonNode.get("resultTable").get("rows").get(0).get(1);
+
+    Pattern pattern = Pattern.compile("LogicalValues\\(tuples=\\[\\[]]\\)");
+    String planAsText = plan.asText();
+    boolean matches = pattern.matcher(planAsText).find();
+    Assert.assertFalse(matches, "Plan should not contain contain LogicalValues node but plan is \n"
+        + planAsText);
```

Comment:
Why not run an actual query instead and assert that the size of the result row set is non-zero?

Reply:
I am running the query alter, but this ensures the planning is correct. In case we have a regression, it may fail in the planning check or in the result check.

If it fails in the plan phase, the error is more clear and therefore easier to fix.

Reply:
> I am running the query alter

Ah, can't believe I missed that, it's right after this. My bad 🤦

Reply:
It can happen to the bests ;)